packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart

```diff
@@ -190,31 +232,25 @@ class ClassesTableDiff extends StatelessWidget {
     ),
   ];
 
-  static final _retainedSizeDeltaColumn =
-      _SizeColumn(_DataPart.delta, _SizeType.retained);
+  final Widget classFilterButton;
 
-  static late final List<ColumnData<DiffClassStats>> _columns =
-      <ColumnData<DiffClassStats>>[
-    _ClassNameColumn(),
-    _InstanceColumn(_DataPart.created),
-    _InstanceColumn(_DataPart.deleted),
-    _InstanceColumn(_DataPart.delta),
-    _SizeColumn(_DataPart.created, _SizeType.shallow),
-    _SizeColumn(_DataPart.deleted, _SizeType.shallow),
-    _SizeColumn(_DataPart.delta, _SizeType.shallow),
-    _SizeColumn(_DataPart.created, _SizeType.retained),
-    _SizeColumn(_DataPart.deleted, _SizeType.retained),
-    _retainedSizeDeltaColumn,
-  ];
+  static final _columnStore = <String, _ClassesTableDiffColumns>{};
+  static _ClassesTableDiffColumns _columns(
+    Widget classFilterButton,
+  ) =>
+      _columnStore.putIfAbsent(
+        '${identityHashCode(classFilterButton)}',
+        () => _ClassesTableDiffColumns(classFilterButton),
+      );
```

Comment:
this is confusing. Why do we have a column store where we store a set of columns based on a widget's hash code?

Reply:
replaced with snapshotId